src/structures/member.ts

```diff
@@ -60,44 +68,35 @@ const baseMember: Partial<Member> = {
 };
 
 export async function createMemberStruct(
-  data: MemberCreatePayload,
+  data: DiscordGuildMember,
   guildId: string,
 ) {
   const {
-    joined_at: joinedAt,
-    premium_since: premiumSince,
-    user: userData,
-    roles,
-    deaf,
-    mute,
-    nick,
+    user,
+    joinedAt,
+    premiumSince,
     ...rest
-  } = data;
+  } = snakeKeysToCamelCase(data) as GuildMember;
 
-  const { mfa_enabled: mfaEnabled, premium_type: premiumType, ...user } =
-    data.user || {};
-
-  const restProps: Record<string, ReturnType<typeof createNewProp>> = {};
+  const props: Record<string, ReturnType<typeof createNewProp>> = {};
 
   for (const key of Object.keys(rest)) {
     // @ts-ignore index signature
-    restProps[key] = createNewProp(rest[key]);
+    props[key] = createNewProp(rest[key]);
   }
 
-  for (const key of Object.keys(user)) {
+  for (const key of Object.keys(user!)) {
     // @ts-ignore index signature
-    restProps[key] = createNewProp(user[key]);
+    props[key] = createNewProp(user[key]);
   }
 
   const member = Object.create(baseMember, {
-    ...restProps,
-    mfaEnabled: createNewProp(mfaEnabled),
-    premiumType: createNewProp(premiumType),
+    ...props,
     /** The guild related data mapped by guild id */
     guilds: createNewProp(new Collection<string, GuildMember>()),
   });
 
-  const cached = await cacheHandlers.get("members", user.id);
+  const cached = await cacheHandlers.get("members", user!.id);
```

Comment:
are u certain it will never be undefinewd here?

Reply:
yes 
> The field `user` won't be included in the member object attached to `MESSAGE_CREATE` and `MESSAGE_UPDATE` gateway events.